bob-backend/src/pearl/group.rs

```diff
@@ -330,9 +350,16 @@ impl Group {
         )
     }
 
-    pub fn create_pearl_by_timestamp(&self, time: u64) -> Holder {
-        let start_timestamp =
-            Stuff::get_start_timestamp_by_timestamp(self.settings.timestamp_period(), time);
+    pub fn create_pearl_by_timestamp(
+        &self,
+        time: u64,
+        timestamp_config: &StartTimestampConfig,
+    ) -> Holder {
+        let start_timestamp = Stuff::get_start_timestamp_by_timestamp(
```

Comment:
@pyakushin please, check carefully that the new logic for setting timestamps does not break anything

Reply:
Checked, everything is ok, behavior changed only for alien puts.